pex/pex_builder.py

```diff
@@ -224,6 +226,58 @@ def set_script(self, script):
         'Could not find script %r in any distribution %s within PEX!' % (
             script, ', '.join(str(d) for d in self._distributions)))
 
+  def _verify_entry_point(self):
+    """
+    Given entry_point `a.b.c:m`, make sure
+    1. a/b/c.py exists
+    2. `m` method is defined in a/b/c.py
+
+    :param entry_point:
+    :return:
+    """
+
+    entry_point = self._pex_info.entry_point
+    ep_split = entry_point.split(':')
+
+    # a.b.c:m ->
+    # ep_module = 'a.b.c'
+    # ep_method = 'm'
+
+    # Only module is specified
+    if len(ep_split) == 1:
+      ep_module = ep_split[0]
+      import_statement = 'import {}'.format(ep_module)
+    elif len(ep_split) == 2:
+      ep_module = ep_split[0]
+      ep_method = ep_split[1]
+      import_statement = 'from {} import {}'.format(ep_module, ep_method)
+    else:
+      raise self.InvalidEntryPoint("Fail to parse: `{}`".format(entry_point))
+
+    args = [self._interpreter.binary, '-c', import_statement]
```

Comment:
Execution via subprocess was consolidated to [`Executor`](https://github.com/pantsbuild/pex/blob/c07c70f31d8070f26f45b31f84f84741ed2593a1/pex/executor.py#L87-L98), so this code should use that instead of `subprocess` directly.

A bit more true to the runtime environment the entry point will be executed in would be something like:
```python
with named_temporary_file() as fp:
  fp.write(import_statement)
  fp.close()
  result = PEX(self.chroot().path()).run([fp.name], env={'PEX_INTERPRETER': '1'})
  if result != 0:
    raise self.InvalidEntryPoint('Failed to validate entrypoint {}'.format(entry_point))
```
This though unfortunately creates a circular PEX <-> PEXBuilder dep. Better I think to lift the check to the PEX constructor which could be plumbed both from pex.bin.pex and from API users (Pants).